### PR TITLE
chore: add binary validation in all builds

### PR DESCRIPTION
**Pull request by sobolk**
#### Description of changes

**This PR solves the following:**
- Builds all binaries in every build including pull requests. I.e. this can be caught earlier than in e2e tests https://github.com/aws-amplify/amplify-cli/pull/11371 .
- It verifies all binaries sizes against hardcoded threshold.
- Parallelizes binary builds, so it's no longer needed to skip some to save time. And to make sure PR build time is kept same.

**PR validation**

Before:
image

After:
image

**E2E tests**

Before:
image

After:
image

**Verification step**
image

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] [Pull request labels](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#labels) are added

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by awsluja** [comment]
Can you provide a negative & negative run? or at least confirm it fails the job if file size is over 1mb (for example)

File: .circleci/local_publish_helpers.sh [hunk 1]
```diff
@@ -78,23 +72,65 @@ function generatePkgCli {
 
   # Build pkg cli
   cp package.json ../build/node_modules/package.json
-  if [[ "$CIRCLE_BRANCH" == "release" ]] || [[ "$CIRCLE_BRANCH" =~ ^run-e2e-with-rc\/.* ]] || [[ "$CIRCLE_BRANCH" =~ ^release_rc\/.* ]] || [[ "$CIRCLE_BRANCH" =~ ^tagged-release ]]; then
-    # This will generate a file our arm64 binary
+
+  if [[ "$@" =~ 'arm' ]]; then
     npx pkg --no-bytecode --public-packages "*" --public -t node14-linux-arm64 ../build/node_modules -o ../out/amplify-pkg-linux-arm64
```
Comment (akshbhu) [inline .circleci/local_publish_helpers.sh:77]: can you try removing `--public` flag and check for the arm binary

File: .circleci/local_publish_helpers.sh [hunk 2]
```diff
@@ -78,23 +72,65 @@ function generatePkgCli {
...
   cd ..
 }
+
+function verifyPkgCli {
+    echo "Human readable sizes"
+    du -h out/*
+    echo "Sizes in bytes"
+    wc -c out/*
+
+    linux_compressed_binary_threshold_in_bytes=$((200 * 1024 * 1024))
```
Comment (akshbhu) [inline .circleci/local_publish_helpers.sh:105]: default size is chosen based on previous data ?

**Review by akshbhu — APPROVED** [review]
lgtm

File: .circleci/local_publish_helpers.sh (hunk 2 above)
Comment (sobolk) [reply .circleci/local_publish_helpers.sh:105]: By looking at current state of art. I.e. here.
image

File: .circleci/local_publish_helpers.sh (hunk 1 above)
Comment (sobolk) [reply .circleci/local_publish_helpers.sh:77]: I can but not here. This is not the goal of this PR.

**Review by jhockett — APPROVED** [review]

**Pull request merged**